files/setup_sudoers.sh

```diff
@@ -0,0 +1,55 @@
+#! /bin/bash
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Read the project metadata key named "sudoers" and add each comma separated value to
+# the sudoers file.
+# +TODO - add switch to permit fetching sudoers key from instance metadata
+stdlib::setup_sudoers() {
+  local user user_list sudoers_file
+  user_list="$(stdlib::metadata_get -k 'project/attributes/sudoers')"
+
+  if [[ -z "${user_list}" ]]; then
+    stdlib::debug "Skipping sudoers setup.  The value of the project metadata key named sudoers is empty.  Set sudoers to a comma separated list to enable sudo support, e.g. sudoers=jmccune,pames"
+    return 0
+  fi
+
+  sudoers_file="$(mktemp)"
+  cp /etc/sudoers "${sudoers_file}"
+
+  for user in ${user_list//,/ }; do
+    if grep -q "^${user}"'\b' "${sudoers_file}"; then
+      stdlib::debug "User ${user} is already in /etc/sudoers, taking no action"
+    else
+      stdlib::info "Adding ${user} to /etc/sudoers"
+      echo -e "${user}\tALL= (ALL)\tNOPASSWD: ALL" >>"${sudoers_file}"
```

Comment:
Have we considered making `NOPASSWD` optional? I believe not all users would want this option enabled.

Reply:
Good idea.  I did originally, yes, but applied YAGNI.  Suggest watching out for this feature request from the field.

Reply:
Also note the default behavior on GCP results in users created without passwords.

Reply:
So, I'm committing changes that permit his behavior -- pass a switch during the invocation of the `setup_sudoers` function in `startup-script-custom`,

 _However_, I'm not convinced this belongs in this PR. 

This opens up a wider question and this functionality only satisfies the use case when users are managed _outside_ OSLogin. Default behavior is that all user accounts in OSLogin get added to the `google-sudoers` unix group with the `NOPASSWD` Tag_spec. See the `/etc/sudoers.d/google_sudoers` file on any box launched.

Reply:
That's fair. We can forgo adding this for now.